File: bmodle/src/main/java/com/hong_world/bmodle/modle/BaseTasksDataSource.java
package com.hong_world.bmodle.modle;

import com.hong_world.common.bean.FeedArticleListData;

import io.reactivex.Observable;

/**
 * Date: 2018/8/13. 10:44
 * Author: hong_world
 * Description:
 * Version:
 */
public class BaseTasksDataSource implements TasksDataSource {
    protected static BaseTasksDataSource INSTANCE;

    protected BaseTasksDataSource() {
    }

    public static BaseTasksDataSource getInstance() {
        if (INSTANCE == null) {
            INSTANCE = new BaseTasksDataSource();
        }
        return INSTANCE;
    }

    @Override
    public Observable<FeedArticleListData> getFeedArticleList(int num) {
        return null;
    }
}

File: kotlin-module/src/main/java/com/hong_world/kotlin_module/modle/BaseTasksDataSource.kt
package com.hong_world.kotlin_module.modle

import com.hong_world.kotlin_module.bean.FeedArticleListData
import io.reactivex.Observable

/**
 * Date: 2018/9/25. 16:22
 * Author: hong_world
 * Description:
 * Version:
 */
open class BaseTasksDataSource : TasksDataSource {


    protected constructor()

//    companion object {
//        @JvmStatic
//        fun getInstance() = Holder.INSTANCE
//    }
//
//    protected object Holder {
//        val INSTANCE = BaseTasksDataSource()
//    }

    override fun getFeedArticleList(num: Int): Observable<FeedArticleListData>? = null
}